### PR TITLE
[SQSERVICES-1773] Allow pagination for team search endpoints - `has_more`

### PR DESCRIPTION
**Pull request by battermann**
This is an extension of https://github.com/wireapp/wire-server/pull/2895 and adds the `has_more` field to the search result.

https://wearezeta.atlassian.net/browse/SQSERVICES-1773

## Checklist

 - [x] No new changelog entry needed as this already exists in the base PR
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: services/brig/src/Brig/User/Search/TeamUserSearch.hs [hunk 1]
```diff
@@ -50,12 +50,12 @@ teamUserSearch ::
...
   Maybe PagingState ->
   m (SearchResult TeamContact)
-teamUserSearch tid mbSearchText mRoleFilter mSortBy mSortOrder (fromRange -> s) mPagingState = liftIndexIO $ do
+teamUserSearch tid mbSearchText mRoleFilter mSortBy mSortOrder (fromRange -> size) mPagingState = liftIndexIO $ do
   let (IndexQuery q f sortSpecs) = teamUserSearchQuery tid mbSearchText mRoleFilter mSortBy mSortOrder
   idx <- asks idxName
   let search =
         (ES.mkSearch (Just q) (Just f))
-          { ES.size = ES.Size (fromIntegral s),
+          { ES.size = ES.Size (fromIntegral size + 1),
```
Comment (mdimjasevic) [inline services/brig/src/Brig/User/Search/TeamUserSearch.hs:58]: Why is it + 1?

**Review by mdimjasevic — APPROVED** [review]
Please add a changelog. You can extend the one from the previous PR by listing both PR numbers, which should look nice in the release notes.

The part I didn't get is +1; a comment is inlined.

Furthermore, is there a way for the API user to iterate over result pages?

**Comment by battermann** [comment]
> Please add a changelog. You can extend the one from the previous PR by listing both PR numbers, which should look nice in the release notes.
> 
> The part I didn't get is +1; a comment is inlined.

Good points, I'll add a comment about the +1 and update the changelog.
 
> Furthermore, is there a way for the API user to iterate over result pages?

Not sure what you mean exactly. This PR adds the `has_more` field. If it is true, it means that there is a next page. They can use the `paging_state` and include it in a request, and this will give them the next page.

**Comment by mdimjasevic** [comment]
> 
> > Furthermore, is there a way for the API user to iterate over result pages?
> 
> Not sure what you mean exactly. This PR adds the `has_more` field. If it is true, it means that there is a next page. They can use the `paging_state` and include it in a request, and this will give them the next page.

Ok, that's what I was asking about. Thanks for clarifying.

Given the experience with the Cassandra paging endpoints, I expect client devs to be asking how to use this Es paging system. Perhaps it'd be good to include a thorough description in Swagger analogous to the way the Cassandra paging endpoints have it, by extending the endpoint description.

**Comment by battermann** [comment]
> Given the experience with the Cassandra paging endpoints, I expect client devs to be asking how to use this Es paging system. Perhaps it'd be good to include a thorough description in Swagger analogous to the way the Cassandra paging endpoints have it, by extending the endpoint description.

I updated the docs, and tried to be more clear. From the client perspective, ES's and C*'s paging mechanisms works exactly the same. In other words, the API is the same (only the implementation differs). So I assume they are already fairly familiar with it. However, it can't hurt to be more clear.

**Pull request merged**